src/protos/grpc/master_metadata_service.proto

```diff
@@ -0,0 +1,81 @@
+syntax = "proto3";
+
+package protos.grpc;
+
+import "google/protobuf/empty.proto";
+import "src/protos/metadata.proto";
+
+service MasterMetadataService {
+  // Create a new file chunk for |filename|, and return the associated
+  // chunk index and metadata. If |filename| already exists, it creates and
+  // returns a new file chunk for |filename|. If |filename| doesn't exist,
+  // it creates the file and returns the first uninitialized empty chunk.
+  rpc CreateFileChunk(CreateFileRequest) returns (CreateFileReply) {}
+
+  // Return the chunk handle, version, and associated replica locations 
+  // for a given |filename| at |chunk_index|. If either |filename| or 
+  // |chunk_index| doesn't exist, an error status with empty reply is returned.
+  rpc OpenFile(OpenFileRequest) returns (OpenFileReply) {}
+
+  // Delete |filename|. This effectively deletes all associated file chunks.
+  // If |filename| doesn't exist yet, an error status is returned.
+  rpc DeleteFile(DeleteFileRequest) returns (google.protobuf.Empty) {}
+
+  // TODO(tugan): add gRPC calls for snapshot operations
+}
+
+message CreateFileRequest {
+  // Absolute file name or directory name
+  string filename = 1;
+}
+
+message CreateFileReply {
+  // The original request associated with this reply.
+  CreateFileRequest request = 1;
+  // A chunk index is floor[byte_range / chunk block size]
+  // The default chunk block size is 64MB.
+  int32 chunk_index = 2;
+  // The metadata associated with the file chunk at this chunk index
+  protos.FileChunkMetadata metadata = 3;
+}
+
+message OpenFileRequest {
+  // Absolute file name or directory name
+  string filename = 1;
+  // A chunk index is floor[byte_range / chunk block size] 
+  // By default, the block size of a chunk is 64MB.
+  int32 chunk_index = 2;
```

Comment:
nit: uint32

Reply:
There are several places in this PR where we should consider using uint32. Don't want to litter this PR with comment.